tests/thread/enc/thread.h

```diff
@@ -19,11 +19,22 @@ static __inline pthread_mutex_t __mutex_initializer_default()
     return m;
 }
 
+static __inline pthread_mutex_t __mutex_initializer_recursive()
+{
+    pthread_mutex_t m;
+    pthread_mutexattr_t attr;
+    pthread_mutexattr_init(&attr);
+    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
+    pthread_mutex_init(&m, &attr);
+    return m;
+}
+
 typedef pthread_t oe_thread_t;
 #define oe_thread_self pthread_self
 
 typedef pthread_mutex_t oe_mutex_t;
-#define OE_MUTEX_INITIALIZER __mutex_initializer_default()
+#define OE_MUTEX_INITIALIZER __mutex_initializer_recursive()
```

Comment:
Do we need these redefinitions? They are already defined in internal/thread.h

Reply:
My understanding is that this header is used by the enclave code so that we can use the same to test pthread_mutex and oe_mutex APIs (by including the header or not).